update version 0.10.4

JamesRobertsonGames opened the pull request:
[PKG-1666]

[PKG-1666]: https://anaconda.atlassian.net/browse/PKG-1666?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

JamesRobertsonGames committed:
update version

JamesRobertsonGames committed:
change sha to trigger pipeline to check aggregate

JamesRobertsonGames committed:
change sha

JamesRobertsonGames committed:
update sha to proc ci update

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:32: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:35: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:36: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:36: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:37: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:38: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:39: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:40: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:59: has_run_test_and_commands: Test commands are not executed when run_test.sh (.bat...) is present.
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:71: http_url: http://igraph.org/c/ is not https
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:71: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:71: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:72: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:72: http_url: http://igraph.org/c/ is not https
ERROR: /tmp/abs_4fr0dj2qhh/clone/recipe/meta.yaml:72: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

JamesRobertsonGames committed:
rectify linting issues

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:32: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:33: host_section_needs_exact_pinnings: Packages in host must have exact version pinnings, except python build tools.
ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_2d9p6rzq3r/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
Errors were found

</details>

JamesRobertsonGames committed:
version lock

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_c4gsz1lbmm/clone/recipe/meta.yaml:68: missing_dev_url: The recipe is missing a dev_url
ERROR: /tmp/abs_c4gsz1lbmm/clone/recipe/meta.yaml:68: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_c4gsz1lbmm/clone/recipe/meta.yaml:69: missing_documentation: The recipe is missing a doc_url or doc_source_url
ERROR: /tmp/abs_c4gsz1lbmm/clone/recipe/meta.yaml:69: missing_dev_url: The recipe is missing a dev_url
Errors were found

</details>

JamesRobertsonGames committed:
add urls to the file

akabanovs committed:
Adding abs.yaml and skippin ppc

akabanovs committed:
A number of changes:
 - Skip s390x and ppc due to unavailability of some deps.
 - Fix host dep versions.
 - Use openblas instead of blas in host and run.
 - Update build.sh to use openblas.

akabanovs committed:
Add all deps to use openblas on non-win and mkl on win.

akabanovs committed:
Do we need python on windows?

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -27,24 +28,31 @@ requirements:
-    - libblas
-    - liblapack
+    - glpk 4.65
+    - libxml2 2.10
```
```suggestion
    - libxml2 {{ libxml2 }}
```
As we have it in the generic cbc.yaml

akabanovs replied:
yes, thanks

akabanovs committed:
Further test for win

akabanovs committed:
remove vs compiler

akabanovs committed:
remove vs compiler from tests

akabanovs committed:
Further tests to build win and osx-64

akabanovs committed:
Try osx-64 with newer SDK

akabanovs committed:
remove local cbc file

akabanovs committed:
Try specifying version of mkl

akabanovs committed:
Deactivate tests on osx-64 and improve mkl pinnings.

akabanovs committed:
Modify mkl pinnings

JeanChristopheMorinPerso commented on recipe/meta.yaml:
```diff
@@ -23,28 +22,33 @@ requirements:
-    - liblapack
+    - glpk 4.65
+    - libxml2 {{ libxml2 }}
+    - suitesparse 7.0.1
```
Conda-builds complains and says that it's not used:
```
   WARNING (igraph): run-exports library package pkgs/main::suitesparse-7.0.1-ha6b1e0b_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)    
```
And looking at the code, I'm not even sure how it would work. They include `cs/cs.h` (see https://github.com/igraph/igraph/blob/0.10.4/src/core/sparsemat.c#L38), but our `suitesparse` package only exposes a `cs.h` file (withough a `cs` folder`...

What's more, the logs are saying
```
[1/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_add.c.o
[2/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_chol.c.o
[3/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_amd.c.o
[4/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_cholsol.c.o
[5/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_compress.c.o
[6/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_cumsum.c.o
[7/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_counts.c.o
[8/344] Building C object vendor/cs/CMakeFiles/cxsparse_vendored.dir/cs_dfs.c.o
```
And on top of that, cmake says that `DIGRAPH_USE_INTERNAL_CXSPARSE` is not used.

sumit0190 replied:
`suitesparse` is no longer a dependency for this package, so yes, it can be removed from the YAML file (along with its flag in the build script).

akabanovs replied:
Thanks

sumit0190 commented on recipe/meta.yaml:
```diff
@@ -70,6 +70,8 @@ about:
     igraph is a collection of network analysis tools with the emphasis on efficiency,
     portabiliy and ease of use. igraph is open source and free. igraph can be
     programmed in R, Python and C/C++."
+  doc_url: https://igraph.org/r/doc/
```
This is pointing to the docs for the R bindings.

akabanovs committed:
Remove suitesparse; win should use our arpack rather than internal; gmp for win too

akabanovs committed:
Use internal gmp on win

akabanovs committed:
Allow building on linux-s390x

akabanovs committed:
fix s390x and hopefully win too

anaconda-pkg-build commented:
<details><summary>Linter check found the following problems:</summary>
The following problems have been found:

ERROR: /tmp/abs_08idcsl0l9/clone/recipe/meta.yaml:15: yaml_load_failure: The recipe could not be loaded by yaml
ERROR: /tmp/abs_08idcsl0l9/clone/recipe/meta.yaml:16: yaml_load_failure: The recipe could not be loaded by yaml
ERROR: /tmp/abs_08idcsl0l9/clone/recipe/meta.yaml:17: yaml_load_failure: The recipe could not be loaded by yaml
Errors were found

</details>

akabanovs committed:
fix s390x

skupr-anaconda commented on recipe/meta.yaml:
```diff
@@ -1,20 +1,21 @@
+  sha256: aa5700b58c5f1e1de1f4637ab14df15c6b20e25e51d0f5a260921818e8f02afc
 build:
   skip: true  # [win and vc<14]
+  skip: true  # [linux and ppc64le]
```
We do not need this package on `s390x`. If it requires a lot of effort you can skip it

akabanovs replied:
s390x now builds. The problem was that glpk was unavailable on s390x. So I am using internal one on s390x only, switching based on arch using env vars.

skupr-anaconda commented on recipe/bld.bat:
```diff
@@ -11,13 +11,13 @@ cmake %CMAKE_ARGS% -GNinja ^
       -DCMAKE_INSTALL_LIBDIR=%LIBRARY_LIB% ^
       -DCMAKE_INSTALL_INCLUDEDIR=%LIBRARY_INC% ^
       -DCMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+      -DCMAKE_MODULE_PATH=%LIBRARY_PREFIX%\mingw-w64\lib\pkgconfig ^
```
```suggestion
      -DCMAKE_MODULE_PATH=%LIBRARY_PREFIX%/mingw-w64/lib/pkgconfig ^
```
Use forward slashes `/` in your paths, see https://stackoverflow.com/a/13742066.
I think it will unblock win64

akabanovs committed:
Another attempt to fix s390x and win builds

akabanovs committed:
Another attempt to fix win build

akabanovs committed:
Attempt to build on win

akabanovs committed:
fix typo

akabanovs committed:
Specify arpack library

akabanovs committed:
Attempt to build on win

akabanovs committed:
Attempt to build on win

akabanovs committed:
Attempt to build on win

akabanovs committed:
Attempt to build on win with mingw

akabanovs committed:
Update bld.bat to try and build win

akabanovs committed:
Use internal arpack on windows

akabanovs committed:
Clean up recipe - forgot to deactivate arpack on win as internal arpack will be used

skupr-anaconda reviewed (approved):
LGTM

akabanovs committed:
Add a flag to upload to staging channel, to test python-igraph

akabanovs committed:
Specify destination channel in abs.yaml file
